### PR TITLE
Level select menu

### PR DESCRIPTION
**Pull request by AndyLoft**

### CODE REVIEW

**Comment by mheyer32** [comment]
Cool!

Do we know what the "Play Game" menu entry was supposed to do?

**Comment by AndyLoft** [comment]
would this level select menu be of use? start any level but with only 20 shotgun shells and 200 health (or what ever the last loaded game slot provided).

**Comment by mheyer32** [comment]
Its definitely good for debugging things. I don't think anyone would mind being able to start anywhere in the game. TBH _I_ haven't seen all levels yet :-D

**Comment by AndyLoft** [comment]
its rather basic but functional.

File: ab3d2_source/controlloop.s [hunk 1]
```diff
@@ -473,6 +482,138 @@ READMAINMENU:
...
+				beq	levelMenu2
+				
+				cmp.w	#0,d0
+				bne.s	.sl2
+				move.w	#0,MAXLEVEL
+				bra	.levelSelectDone
+.sl2
+				cmp.w	#1,d0
+				bne.s	.sl3
+				move.w	#0,MAXLEVEL
```
Comment (0xABADCAFE) [inline ab3d2_source/controlloop.s:518]: I don't know if I am missing something, but aren't all these equivalent to `move.w d0, MAXLEVEL` ? It seems like you could either range check d0 first, or even just mask it with $F and then move it to MAXLEVEL

**Comment by AndyLoft** [comment]
latest push does not work as intended.

File: ab3d2_source/controlloop.s (hunk 1 above)
Comment (AndyLoft) [reply ab3d2_source/controlloop.s]: yeah, still learning.

**Comment by AndyLoft** [comment]
this works but only if the deflev.dat file exists in the level directory, otherwise you get an endless disk prompt loop with the current file.io calls. need a method to check if the file exists and skip to branch to DEFAULTGAME if it fails.

**Comment by AndyLoft** [comment]
latest tests LOADEXT in io.file.s (set in levelMenu: and reset in DEFGAME:) to stop endless disk prompt and branch to DEFAULTGAME if deflev.dat is not found. seems to work but I am sure I have forgotten to do stuff.

**Review by 0xABADCAFE — COMMENTED** [review]
Generally looks OK, but I think it would be neat to make the menu text symbols zero initialised bss data that gets populated from the loaded game data so that you get the proper names.

**Review by 0xABADCAFE — APPROVED** [review]
I think my previous suggestion should be taken out of MVP for this and moved to an enhancement request. To the mergecopter!

**Pull request merged**